File: app/src/main/java/com/nedaluof/qurany/di/modules/DatabaseModule.java
package com.nedaluof.qurany.di.modules;

import android.content.Context;

import com.nedaluof.qurany.data.room.QuranyDatabase;
import com.nedaluof.qurany.data.room.ReciterDao;
import com.nedaluof.qurany.data.room.ReciterSuraDAO;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by nedaluof on 6/26/2020.
 */
@Module
public class DatabaseModule {
    Context context;

    public DatabaseModule(Context context) {
        this.context = context;
    }

    @Singleton
    @Provides
    QuranyDatabase provideDatabase() {
        return QuranyDatabase.getInstance(context);
    }

    @Singleton
    @Provides
    ReciterSuraDAO provideReciterSuraDao(QuranyDatabase database) {
        return database.getReciterSuraDao();
    }

    @Singleton
    @Provides
    ReciterDao provideReciterDao(QuranyDatabase database) {
        return database.getRecitersDao();
    }

}

File: app/src/main/java/com/nedaluof/qurany/di/DatabaseModule.kt
package com.nedaluof.qurany.di

import android.content.Context
import androidx.room.Room
import com.nedaluof.qurany.data.room.QuranyDatabase
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.qualifiers.ApplicationContext
import dagger.hilt.components.SingletonComponent
import javax.inject.Singleton

/**
 * Created by nedaluof on 12/2/2020.
 */
@InstallIn(SingletonComponent::class)
@Module
object DatabaseModule {
    @Singleton
    @Provides
    fun provideDatabase(@ApplicationContext context: Context) =
            Room.databaseBuilder(
                    context,
                    QuranyDatabase::class.java,
                    QuranyDatabase.DB_NAME).build()

    @Singleton
    @Provides
    fun provideReciterDao(database: QuranyDatabase) =
            database.getRecitersDao()


}